### PR TITLE
Convert ML.Sweeper usages of SubComponent to IComponentFactory.

### PR DESCRIPTION
**Pull request by eerhardt**
Moving all the SubComponent usages in the ML.Sweeper assembly to use IComponentFactory.

Note: I logged https://github.com/dotnet/machinelearning/issues/733 for the NelderMeadSweeper, which was using another component that doesn't appear to be in the `dotnet/machinelearning` repo.

Working towards #585

### CODE REVIEW

File: src/Microsoft.ML.Sweeper/AsyncSweeper.cs [hunk 1]
```diff
@@ -193,13 +194,13 @@ public sealed class Arguments
         public DeterministicSweeperAsync(IHostEnvironment env, Arguments args)
         {
             _host = env.Register("DeterministicSweeperAsync", args.RandomSeed);
-            _host.CheckUserArg(args.Sweeper.IsGood(), nameof(args.Sweeper), "Please specify a sweeper");
+            _host.CheckValue(args.Sweeper, nameof(args.Sweeper), "Please specify a sweeper");
```
Comment (codemzs) [inline src/Microsoft.ML.Sweeper/AsyncSweeper.cs]: >CheckValue [](start = 18, length = 10)

why is this changed to CheckValue from CheckUserArgs?

Comment (eerhardt) [reply src/Microsoft.ML.Sweeper/AsyncSweeper.cs]: SubComponents have an IsGood() extension method that check both for null and for a non-empty "Kind" property.  CheckUserArg only has an overload for bool, which IsGood() returns.

Now that this is an IComponentFactory, the only thing we can check for is null - which is what CheckValue does.
